### PR TITLE
Add Request Review functions (list, create, remove)

### PR DESCRIPTION
**Pull request by vglocus**
Extend PullRequestClient with methods for listing, creating and removing requested reviews.

Can reach out using viktorg@spotify.com

### CODE REVIEW

**Comment by vglocus** [comment]
@henriquetruta

File: src/main/java/com/spotify/github/v3/clients/PullRequestClient.java [hunk 1]
```diff
@@ -173,6 +175,46 @@ static PullRequestClient create(
...
+   * Remove a request for review for a pull request.
+   *
+   * @param number pull request number
+   * @param properties properties for reviewing the PR, such as reviewers and team_reviewers.
+   * @see "https://docs.github.com/en/rest/reference/pulls#request-reviewers-for-a-pull-request"
+   */
+  public CompletableFuture<Void> removeRequestedReview(final int number, final RequestReviewParameters properties) {
+    final String path = String.format(PR_REVIEW_REQUESTS_TEMPLATE, owner, repo, number);
+    final String jsonPayload = github.json().toJsonUnchecked(properties);
+    log.debug("Requesting reviews for PR: " + path);
```
Comment (henriquetruta) [inline src/main/java/com/spotify/github/v3/clients/PullRequestClient.java:214]: this comment seems to have been copied from the one above

File: src/main/java/com/spotify/github/v3/clients/PullRequestClient.java [hunk 2]
```diff
@@ -192,4 +234,6 @@ static PullRequestClient create(
     log.debug("Fetching pull requests from " + path);
     return github.request(path, LIST_PR_TYPE_REFERENCE);
   }
+
```
Comment (henriquetruta) [inline src/main/java/com/spotify/github/v3/clients/PullRequestClient.java:237]: minor: unnecessary new lines

File: src/main/java/com/spotify/github/v3/clients/PullRequestClient.java [hunk 3]
```diff
@@ -173,6 +175,46 @@ static PullRequestClient create(
...
+  }
+
+  /**
+   * Remove a request for review for a pull request.
+   *
+   * @param number pull request number
+   * @param properties properties for reviewing the PR, such as reviewers and team_reviewers.
+   * @see "https://docs.github.com/en/rest/reference/pulls#request-reviewers-for-a-pull-request"
+   */
+  public CompletableFuture<Void> removeRequestedReview(final int number, final RequestReviewParameters properties) {
```
Comment (henriquetruta) [inline src/main/java/com/spotify/github/v3/clients/PullRequestClient.java:211]: it'd be nice to add a test for this case, so we can get the new top-level `delete(path, jsonPayload)` method tested as well

**Review by henriquetruta — CHANGES_REQUESTED** [review]
Thanks for your PR! A few suggestions inline

**Comment by vglocus** [comment]
Thank you @henriquetruta. I've fixed your comments and extended the testing somewhat.

**Review by henriquetruta — APPROVED** [review]
Thanks for addressing the comments!

**Pull request merged**